packages/idyll-components/src/heading1.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+class Heading1 extends React.PureComponent {
+  render() {
+    const { children, id } = this.props;
+    if (children && !id) {
+      let generatedId = children
+        .toString()
+        .replace(/\s+/g, '-')
+        .toLowerCase();
+      return <h1 id={generatedId}>{children}</h1>;
+    } else if (children && id) {
+      return <h1 id={id}>{children}</h1>;
+    } else {
+      return null;
+    }
+  }
+}
+
+Heading1._idyll = {
+  name: 'Heading1',
+  tagType: 'open',
+  children: ['My Header'],
+  props: [
+    {
+      name: 'id',
+      type: 'string',
+      example: '"my-id"',
+      description: 'attach id attribute to header'
+    }
+  ]
+};
+
+export default Heading1;
```

Comment:
Can you update this code so that it can produce headers at an arbitrary level? E.g. Create a function that produces a new header component with a specified level 1-6

Reply:
we need to pass extra props with the value of [1-6]
and we can render tag based on that.
what do you think of this approach?

Reply:
I'd prefer not to require the user to do that so that people who are using the syntax like:

```
# Header 1
## Header 2
... etc
```

will still be able to benefit from these updates